runtime/interpreter/interpreter.go

```diff
@@ -4481,50 +4530,63 @@ func (interpreter *Interpreter) pathCapabilityCheckFunction(
 				panic(errors.NewUnreachableError())
 			}
 
-			target, authorized, err :=
```

Comment:
the logic here was moved as-is to the `Interpreter.checkValueAtPath` function, so it can be reused in the `Interpreter.checkValue()` function.